board/main.c

```diff
@@ -758,6 +758,35 @@ void tick_handler(void) {
   TICK_TIMER->SR = 0;
 }
 
+// cppcheck-suppress unusedFunction ; not implemented for H7 yet
+void EXTI_IRQ_Handler(void) {
+  if (check_exti_irq()) {
+    current_board->set_usb_power_mode(USB_POWER_CDP);
+    power_save_status = POWER_SAVE_STATUS_DISABLED;
+    heartbeat_counter = 0U;
+
+    current_board->set_led(LED_GREEN, true);
+    delay(512000U);
+    current_board->set_led(LED_GREEN, false);
+    delay(512000U);
+
+    exti_irq_clear();
+    clock_init();
+  }
+}
+
+void RTC_WKUP_IRQ_Handler(void) {
+  exti_irq_clear();
+  clock_init();
+
+  current_board->set_led(LED_RED, true);
+  delay(512000U);
+  current_board->set_led(LED_RED, false);
+  current_board->set_led(LED_BLUE, true);
+  delay(512000U);
+  current_board->set_led(LED_BLUE, false);
```

Comment:
wouldn't it make more sense to only toggle the LED state and immediately go back to sleep?

Reply:
This PR is still draft ;) Testing different modes. And LED takes ~25mW, so not sure about toggling yet.